src/models/device-type.ts

```diff
@@ -178,8 +180,8 @@ const getDeviceTypeModel = function (deps: InjectedDependenciesParam) {
 							},
 							options,
 						),
-					)
-				)[0];
+					)) as OptionsToResponse<DeviceType['Read'], T, undefined>
+				)[0] as OptionsToResponse<DeviceType['Read'], T, typeof idOrSlug>;
```

Comment:
Am I right that this looks like a bug in pinejs-client typings?
In which case maybe better avoid the double cast and
// @ts-expect-error this should have been working, maybe it's a pinejs-client but or we are missing something 
deviceType = $deviceType